scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -1021,8 +1017,8 @@ protected PskSecretResult requestPskSecretResult(PskPublicInformation pskIdentit
 		String hmacAlgorithm = session.getCipherSuite().getPseudoRandomFunctionMacName();
 		pskRequestPending = true;
 		this.otherSecret = SecretUtil.create(otherSecret);
-		return advancedPskStore.requestPskSecretResult(connection.getConnectionId(), serverNames, pskIdentity,
-				hmacAlgorithm, otherSecret, generateRandomSeed());
+		return advancedPskStore.requestMasterSecret(connection.getConnectionId(), serverNames, pskIdentity,
+				hmacAlgorithm, otherSecret, generateRandomSeed(), resultHandler);
```

Comment:
Now the mechanism has the `resultHandler` as state.
My intention of the API was to allow a stateless implementation (at least no new state).
It would just send away information and any result back could be processed without additional state.
The "other_secret" belongs more to the state, the handshaker needs anyway.

Reply:
I didn't take this stateless requirement into account.
Without it, I think it's pretty common to use callback for async API.

Do you have an exemple of code or a usecase where this stateless requirement is needed ? I have difficulty to see the benenfit right now.

Reply:
Why should such a specific callback be used, if the general one works as well?
There is just no reason to always create a new callback instead of using the single one for the connector.

Reply:
By the way: 
I thought, we have been discussing, if the master secret should have always been returned. I don't see, that this change focus on the difference of returning the master secret. Or do I miss something?

Reply:
1. To avoid the setter in an Interface which is generally not so elegant.
2. We could imagine to not look in the connectionStore when processing result by creating a dedicate callback (but this is not compatible with your stateless requirement)
3. this allow to have a consistent API with [AbstractSyncPskStore](https://github.com/eclipse/californium/blob/async_psk_store/scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/pskstore/AbstractSyncPskStore.java#L51)